zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,395 @@
+             available in the dataset.
+        minute_index : list of integers
+             nanosecond integer representation of the minutes, the enumeration of which
+             corresponds to the values in each bcolz carray.
+        """
+        metadata = {
+            'first_trading_day': str(self.first_trading_day.date()),
+            'minute_index': self._minute_index.asi8.tolist(),
+            'ohlc_ratio': self._ohlc_ratio,
+        }
+        with open(self.metadata_path, 'w+') as fp:
+            json.dump(metadata, fp)
+
+    def sidpath(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        out : string
+            Full path to the bcolz rootdir for the given sid.
+        """
+        sid_subdir = _sid_subdir_path(sid)
+        return join(self._rootdir, sid_subdir)
+
+    def last_date_in_output_for_sid(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        out : pd.Timestamp
+            The midnight of the last date written in to the output for the
+            given sid.
+        """
+        try:
+            sizes_path = "{0}/close/meta/sizes".format(self.sidpath(sid))
+            with open(sizes_path, mode='r') as f:
+                sizes = f.read()
+            data = json.loads(sizes)
+            num_days = data['shape'][0] / self._minutes_per_day
+            if num_days == 0:
+                # empty container
+                return None
+            return self._trading_days[num_days - 1]
+        except IOError:
+            return None
+
+    def _init_ctable(self, path):
+        """
+        Create empty ctable for given path.
+
+        Parameters:
+        -----------
+        path : string
+            The path to rootdir of the new ctable.
+        """
+        # Only create the subdir on container creation.
+        sid_dirname = os.path.dirname(path)
+        os.makedirs(sid_dirname)
+        initial_array = np.empty(0, np.uint32)
+        table = ctable(
+            rootdir=path,
+            columns=[
+                initial_array,
+                initial_array,
+                initial_array,
+                initial_array,
+                initial_array,
+            ],
+            names=[
+                'open',
+                'high',
+                'low',
+                'close',
+                'volume'
+            ],
+            expectedlen=self._expectedlen,
+            mode='w',
+        )
+        table.flush()
+
+    def _zerofill(self, table, numdays):
+        num_to_prepend = numdays * self._minutes_per_day
+        prepend_array = np.zeros(num_to_prepend, np.uint32)
+        # Fill all OHLCV with zeros.
+        table.append([prepend_array] * 5)
+        table.flush()
+
+    def write(self, sid, days, df):
```

Comment:
The expected usage pattern here feels weird to me.  If I understand correctly the expectation here is that I do something like:

```
writer = BcolzMinuteWriter(...)
days = all_my_days()
for sid in all_my_sids():
    writer.write(sid, days, data_for_sid(sid))
```
I can't imagine a world where I'd want `days` to be different from call to call here, which suggests to me that it should be a constructor parameter rather than a parameter to `write`.  Note that we don't have the same problem for the daily writers, because the expectation is that in normal usage there's only one call made to `write`.